examples/lb3app/src/application.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/example-todo
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BootMixin} from '@loopback/boot';
+import {ApplicationConfig} from '@loopback/core';
+import {RepositoryMixin} from '@loopback/repository';
+import {
+  RestApplication,
+  OpenAPIObject,
+  OperationObject,
+  RestServer,
+} from '@loopback/rest';
+import {RestExplorerComponent} from '@loopback/rest-explorer';
+import * as path from 'path';
+import {promisify} from 'util';
+import {MySequence} from './sequence';
+
+const legacyApp = require('../../legacy/server/server');
+const legacyBoot = promisify(require('loopback-boot'));
+const {generateSwaggerSpec} = require('loopback-swagger');
+const swagger2openapi = require('swagger2openapi');
+
+export class TodoListApplication extends BootMixin(
+  RepositoryMixin(RestApplication),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    // Set up the custom sequence
+    this.sequence(MySequence);
+
+    // Set up default home page
+    this.static('/', path.join(__dirname, '../../public'));
+
+    this.component(RestExplorerComponent);
+
+    this.projectRoot = __dirname;
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {
+      controllers: {
+        // Customize ControllerBooter Conventions here
+        dirs: ['controllers'],
+        extensions: ['.controller.js'],
+        nested: true,
+      },
+    };
+  }
+
+  async boot() {
+    // Boot the legacy LB3 app first
+    await legacyBoot(legacyApp);
```

Comment:
We probably should add a `Booter` for LB3 applications to make it part of `app.boot()` process.

Reply:
Let's wait until we see the final shape.

I am little bit reluctant to add a Booter component. I see at least three different ways how LB3 users may want to add their existing LB3 app to LB4 and I am not sure if we can come up with a Booter API that will accommodate all different use cases:

- Mount the entire LB3 express application, including any middleware.
- Mount REST handler only (LB3/strong-remoting router).
- Expose LB3 models via LB4 REST layer as shown in https://github.com/strongloop/loopback-next/pull/2274

Reply:
Done.